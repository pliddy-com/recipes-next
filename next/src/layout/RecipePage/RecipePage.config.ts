interface Breakpoints {
  viewMin?: number;
  imgWidth: number;
}

interface Config {
  breakpoints?: Breakpoints[];
}

const config: Config = {
  breakpoints: [
    {
      viewMin: 1048,
      imgWidth: 600
    },
    {
      viewMin: 960,
      imgWidth: 500
    },
    {
      viewMin: 848,
      imgWidth: 900
    },
    {
      viewMin: 748,
      imgWidth: 800
    },
    {
      viewMin: 648,
      imgWidth: 700
    },
    {
      viewMin: 548,
      imgWidth: 600
    },
    {
      viewMin: 448,
      imgWidth: 500
    },
    {
      viewMin: 332,
      imgWidth: 400
    },
    {
      imgWidth: 300
    }
  ]
};

export default config;
