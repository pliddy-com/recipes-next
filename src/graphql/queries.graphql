# Query for Homepage RecipeCollection (all recipes)

query homePageQuery {
  recipeCollection(order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe collection based on filter

query recipeCollection($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    total
    items {
      ...RecipeDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)

query listPageQuery($where: TagFilter) {
  tagCollection(where: $where, limit: 1) {
    total
    items {
      slug
      title
      linkedFrom {
        recipeCollection(limit: 24) {
          total
          items {
            ...RecipeSummary
          }
        }
      }
    }
  }
}

# Query for recipe slugs

query recipeSlugsCollection($where: RecipeFilter) {
  recipeCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query categorySlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query tagSlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
      linkedFrom {
        recipeCollection {
          total
        }
      }
    }
  }
}

# query for Taxonomy with Category & Subcategory levels

query taxonomyCollection($where: TaxonomyFilter) {
  taxonomyCollection(where: $where, limit: 1) {
    items {
      ...TaxonomyDefault

      tag {
        ...TagDefault
      }

      childrenCollection(limit: 24) {
        total
        items {
          ... on Tag {
            ...TagDefault

            linkedFrom {
              recipeCollection(limit: 1) {
                total
              }
            }
          }
          ...TaxonomyWithChildren
        }
      }
    }
  }
}
