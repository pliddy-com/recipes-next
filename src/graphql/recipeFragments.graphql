# Fragments with default fields for Recipe-specific sections:
#   * Ingredients
#   * Instructions
#   * Image
#   * Recipe

fragment IngredientsDefault on IngredientSection {
  sys {
    id
  }
  title
  slug
  label
  ingredientList
}

fragment InstructionsDefault on InstructionSection {
  sys {
    id
  }
  title
  slug
  label
  instructionList
}

fragment ImageDefault on Asset {
  sys {
    id
  }
  __typename
  title
  description
  contentType
  fileName
  size
  url
  height
  width
}

fragment DescriptionDefault on Recipe {
  description {
    json
    links {
      entries {
        inline {
          __typename
          sys {
            id
            __typename
          }
        }
        hyperlink {
          __typename
          sys {
            id
            __typename
          }
        }
        block {
          __typename

          sys {
            id
            __typename
          }
        }
      }
      assets {
        hyperlink {
          __typename
          title
        }
        block {
          __typename
          title
        }
      }
    }
  }
}

fragment RecipeDefault on Recipe {
  sys {
    id
  }
  __typename
  title
  slug
  description {
    json
  }
  abstract
  image {
    ...ImageDefault
  }
  ingredientsCollection(limit: 10) {
    items {
      ...IngredientsDefault
    }
  }
  equipment
  instructionsCollection(limit: 10) {
    items {
      ...InstructionsDefault
    }
  }
  notes
  tagsCollection(limit: 10) {
    items {
      ...TagDefault
    }
  }
}
