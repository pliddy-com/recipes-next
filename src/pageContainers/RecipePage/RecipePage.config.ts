interface Breakpoints {
  viewMin?: number;
  imgWidth: number;
}

interface Config {
  breakpoints?: Breakpoints[];
}

const config: Config = {
  breakpoints: [
    {
      viewMin: 831,
      imgWidth: 900,
    },
    {
      viewMin: 665,
      imgWidth: 750,
    },
    {
      viewMin: 483,
      imgWidth: 600,
    },
    {
      viewMin: 333,
      imgWidth: 450,
    },
    {
      imgWidth: 300,
    },
  ],
};

export default config;
