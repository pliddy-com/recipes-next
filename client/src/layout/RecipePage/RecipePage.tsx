import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import DynamicImage from 'components/Image/DynamicImage/DynamicImage';
import EquipmentSection from 'components/RecipeSections/EquipmentSection/EquipmentSection';
import IngredientsSection from 'components/RecipeSections/IngredientsSection/IngredientsSection';
import InstructionsSection from 'components/RecipeSections/InstructionsSection/InstructionsSection';
import NotesSection from 'components/RecipeSections/NotesSection/NotesSection';
import RichText from 'components/RichText/RichText';
import TagsSection from 'components/RecipeSections/TagsSection/TagsSection';

import { minToTime } from 'lib/utils';

import { RecipeDefaultFragment, RecipeDescription } from 'types/queries';

import { recipePageConfig } from 'theme/values/images';

interface RecipePageProps {
  content?: RecipeDefaultFragment;
}

const RecipePage = ({ content }: RecipePageProps) => {
  const {
    cookTime,
    description,
    equipment,
    image,
    ingredientsList,
    instructionsList,
    notes,
    prepTime,
    recipeYield,
    tagsCollection,
    title
  } = content ?? {};

  const { aspectRatio, breakpoints } = recipePageConfig;

  const richText = description as RecipeDescription;

  const { items: tags } = tagsCollection ?? {};

  return content ? (
    <Container className="page recipe" data-testid="page" maxWidth="xl">
      <Box className="recipe" data-testid="RecipePage">
        <Typography variant="h1" gutterBottom>
          {title}
        </Typography>

        <Grid container className="content">
          <Grid item lg={6} className="contentGrid">
            <Stack className="description">
              {/* TODO: fix this with a rich text fragment */}
              {description && <RichText content={richText} />}
              {tags && <TagsSection tags={tags} />}

              <Grid container className="details">
                <Grid item xs={6}>
                  <Stack direction="column">
                    {prepTime && (
                      <Typography variant="subtitle2">
                        <strong>Prep Time:</strong>{' '}
                        {minToTime(Number(prepTime))}
                      </Typography>
                    )}
                    {cookTime && (
                      <Typography variant="subtitle2">
                        <strong>Cook Time:</strong>{' '}
                        {minToTime(Number(cookTime))}
                      </Typography>
                    )}
                    {(prepTime || cookTime) && (
                      <Typography variant="subtitle2">
                        <strong>Total Time:</strong>{' '}
                        {minToTime(Number(prepTime) + Number(cookTime))}
                      </Typography>
                    )}
                  </Stack>
                </Grid>
                <Grid item xs={6} className="yield">
                  {recipeYield && (
                    <Typography variant="subtitle2">
                      <strong>Yield:</strong> {recipeYield} servings
                    </Typography>
                  )}
                </Grid>
              </Grid>
            </Stack>
          </Grid>
          <Grid item lg={6}>
            {breakpoints && image && (
              <Box className="image">
                <DynamicImage
                  aspectRatio={aspectRatio}
                  breakpoints={breakpoints}
                  image={image}
                  preload={true}
                />
              </Box>
            )}
          </Grid>
        </Grid>

        <Stack direction="column" spacing={3}>
          {ingredientsList && <IngredientsSection sections={ingredientsList} />}

          {equipment && <EquipmentSection equipment={equipment} />}

          {instructionsList && (
            <InstructionsSection sections={instructionsList} />
          )}

          {notes && <NotesSection notes={notes} />}
        </Stack>
      </Box>
    </Container>
  ) : null;
};

export default RecipePage;
