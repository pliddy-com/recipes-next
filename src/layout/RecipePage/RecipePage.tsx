import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import DynamicImage from 'components/Image/DynamicImage/DynamicImage';
import EquipmentSection from 'components/RecipeSections/EquipmentSection/EquipmentSection';
import IngredientsSection from 'components/RecipeSections/IngredientsSection/IngredientsSection';
import InstructionsSection from 'components/RecipeSections/InstructionsSection/InstructionsSection';
import NotesSection from 'components/RecipeSections/NotesSection/NotesSection';
import RichText from 'components/RichText/RichText';
import TagsSection from 'components/RecipeSections/TagsSection/TagsSection';

import { minToTime } from 'lib/utils';

import { RecipeDefaultFragment, RecipeDescription } from 'types/queries';

import config from './RecipePage.config';

interface RecipePageProps {
  content?: RecipeDefaultFragment;
}

const RecipePage = ({ content }: RecipePageProps) => {
  const {
    cookTime,
    description,
    equipment,
    image,
    ingredientsCollection,
    instructionsCollection,
    notes,
    prepTime,
    recipeYield,
    tagsCollection,
    title,
  } = content ?? {};

  const { breakpoints } = config;

  // TODO: pass full description text into rich text component
  //       & destructure json & links inside component (needs types for json & links)
  //       Need to remove typecasting of RecipeDescription

  // const { json, links } = description ?? {};

  const { items: ingredientsSections } = ingredientsCollection ?? {};
  const { items: instructionsSections } = instructionsCollection ?? {};
  const { items: tags } = tagsCollection ?? {};

  return content ? (
    <Container className="page recipe" data-testid="page" maxWidth="xl">
      <Box className="recipe" data-testid="RecipePage">
        <Typography variant="h1" gutterBottom>
          {title}
        </Typography>

        <Grid container className="content">
          <Grid item lg={6}>
            <Stack className="description">
              {description && (
                <RichText content={description as RecipeDescription} />
              )}
              {tags && <TagsSection tags={tags} />}

              <Grid container className="details">
                <Grid item xs={6}>
                  <Stack direction="column">
                    {prepTime && (
                      <Typography variant="subtitle2">
                        <strong>Prep Time:</strong>{' '}
                        {minToTime(Number(prepTime))}
                      </Typography>
                    )}
                    {cookTime && (
                      <Typography variant="subtitle2">
                        <strong>Cook Time:</strong>{' '}
                        {minToTime(Number(cookTime))}
                      </Typography>
                    )}
                    {(prepTime || cookTime) && (
                      <Typography variant="subtitle2">
                        <strong>Total Time:</strong>{' '}
                        {minToTime(Number(prepTime) + Number(cookTime))}
                      </Typography>
                    )}
                  </Stack>
                </Grid>
                <Grid item xs={6} className="yield">
                  {recipeYield && (
                    <Typography variant="subtitle2">
                      <strong>Yield:</strong> {recipeYield} servings
                    </Typography>
                  )}
                </Grid>
              </Grid>
            </Stack>
          </Grid>
          <Grid item lg={6}>
            {breakpoints && image && (
              <Box className="image">
                <DynamicImage
                  image={image}
                  breakpoints={breakpoints}
                  preload={true}
                />
              </Box>
            )}
          </Grid>
        </Grid>

        <Stack direction="column" spacing={3}>
          {ingredientsSections && (
            <IngredientsSection sections={ingredientsSections} />
          )}

          {equipment && <EquipmentSection equipment={equipment} />}

          {instructionsSections && (
            <InstructionsSection sections={instructionsSections} />
          )}

          {notes && <NotesSection notes={notes} />}
        </Stack>
      </Box>
    </Container>
  ) : null;
};

export default RecipePage;
