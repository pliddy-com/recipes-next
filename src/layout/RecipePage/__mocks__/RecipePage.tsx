import { RecipeDefaultFragment } from 'types/queries';

jest.createMockFromModule('../RecipePage');

export const RecipePage = jest.fn(
  ({ content }: { content: RecipeDefaultFragment }) => {
    const { title, slug, abstract, image } = content;
    const { fileName } = image ?? {};

    return (
      <div data-testid="RecipePage">
        <h1>{title}</h1>
        <p>{slug}</p>
        <p>{abstract}</p>
        <p>{fileName}</p>
      </div>
    );
  }
);

export default RecipePage;
