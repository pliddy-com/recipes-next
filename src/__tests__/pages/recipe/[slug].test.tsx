// import testing-library methods
import { render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import RecipePage, {
  getStaticPaths,
  getStaticProps,
} from 'pages/recipe/[slug]';

import { RecipeDefaultFragment } from 'types/generated/graphql';

// import config object to mock
const config = jest.requireMock('lib/config');

// set up default mock config object
jest.mock('lib/config', () => ({
  microcopy: {
    recipe: {
      defaultTitle: 'Default Index Title',
      defaultDescription: 'Index page description',
    },
    site: {
      title: 'Site Title',
    },
  },
  images: {
    props: {
      recipe: [
        [
          {
            viewMin: 1000,
            imgWidth: 400,
          },
          {
            viewMin: 500,
            imgWidth: 200,
          },
          {
            imgWidth: 100,
          },
        ],
      ],
    },
  },
}));

const recipeContentData = { content: 'recipe content' };
const pageSlugData = { slug: 'slug-1' };

// import api library to mock
const api = jest.requireMock('lib/api');

jest.mock('lib/api', () => ({
  // Cannot access 'pageSlugData' before initialization
  queryPageSlugs: jest.fn().mockResolvedValue(['slug-1']),
  queryRecipeContent: jest
    .fn()
    .mockResolvedValue([{ content: 'recipe content' }]),
}));

describe('RecipePage in recipe/[slug].tsx', () => {
  // reset mocks after each test
  afterEach(() => {
    jest.resetModules();
  });

  describe('when there is page content', () => {
    const pageContent: RecipeDefaultFragment = {
      sys: {
        id: 'sysid-0',
        __typename: 'Sys',
      },
      __typename: 'Recipe',
      title: 'Recipe Title',
      slug: 'recipe-title',
      description: {
        json: {
          data: {},
          content: [
            {
              data: {},
              content: [
                {
                  data: {},
                  marks: [],
                  value: 'Description rich text',
                  nodeType: 'text',
                },
              ],
              nodeType: 'paragraph',
            },
          ],
          nodeType: 'document',
        },
        links: {
          assets: {
            block: [],
            hyperlink: [],
            __typename: 'RecipeDescriptionAssets',
          },
          entries: {
            inline: [],
            hyperlink: [],
            __typename: 'RecipeDescriptionEntries',
          },
          __typename: 'RecipeDescriptionLinks',
        },
        __typename: 'RecipeDescription',
      },
      abstract: 'Recipe abstract',
      image: {
        sys: {
          id: 'sysid-1',
          __typename: 'Sys',
        },
        __typename: 'Asset',
        title: 'Image Title',
        description: 'Image description.',
        contentType: 'image/jpeg',
        fileName: 'filename.jpg',
        size: 99999,
        url: 'https://recipes.pliddy.com/url',
        height: 300,
        width: 400,
      },
      ingredientsCollection: {
        items: [
          {
            sys: {
              id: 'sysid-2',
              __typename: 'Sys',
            },
            title: 'Ingredients Title',
            slug: 'ingredients-title',
            label: 'Ingredients',
            ingredientList: ['ingredient 1', 'ingredient 2'],
            __typename: 'IngredientSection',
          },
        ],
        __typename: 'RecipeIngredientsCollection',
      },
      equipment: ['equipment 1', 'equipment 2'],
      instructionsCollection: {
        items: [
          {
            sys: {
              id: 'sysid-3',
              __typename: 'Sys',
            },
            title: 'Instructions Title',
            slug: 'instructions-title',
            label: 'Label',
            instructionList: ['Instructions item 1'],
            __typename: 'InstructionSection',
          },
        ],
        __typename: 'RecipeInstructionsCollection',
      },
      notes: ['note 1'],
      tagsCollection: {
        items: [
          {
            sys: {
              id: 'sysid-4',
              __typename: 'Sys',
            },
            __typename: 'Tag',
            title: 'Tag 1',
            slug: 'tag-1',
          },
        ],
        __typename: 'RecipeTagsCollection',
      },
    };

    it('it renders the page', async () => {
      const propsContext = {
        preview: false,
        params: pageSlugData,
      };

      const expectedProps = {
        props: { pageContent: recipeContentData, preview: false },
      };

      const expectedPaths = {
        fallback: false,
        paths: [{ params: pageSlugData }],
      };

      const queryPageSlugsSpy = jest.spyOn(api, 'queryPageSlugs');
      const queryRecipeContentSpy = jest.spyOn(api, 'queryRecipeContent');

      const { container } = render(
        <RecipePage pageContent={pageContent} preview={false} />
      );

      expect(await getStaticProps(propsContext)).toEqual(expectedProps);
      expect(await getStaticPaths({})).toEqual(expectedPaths);

      expect(queryPageSlugsSpy).toHaveBeenCalled();
      expect(queryRecipeContentSpy).toHaveBeenCalled();

      // assert that the component matches the existing snapshot
      expect(container).toMatchSnapshot();

      // assert that page head tags are rendered
      const titleTag = document.getElementsByTagName('title')[0];
      expect(titleTag).toBeInTheDocument();
      expect(titleTag.text.includes(pageContent.title as string));

      // test if content is rendered
      const recipe = document.querySelector('.page');
      expect(recipe).toBeInTheDocument();
    });

    describe('when the slug is not a string', () => {
      it('it does not render the page', async () => {
        const propsContext = {
          preview: false,
          params: { slug: undefined },
        };

        try {
          await getStaticProps(propsContext);
        } catch (e) {
          expect(e).toEqual(new Error('Error in SSG!'));
        }
      });
    });
  });

  describe('when there is no page content', () => {
    beforeEach(() => {
      delete config.microcopy;
    });

    it('it does not render the page', () => {
      const pageContent = undefined as unknown as RecipeDefaultFragment;

      render(<RecipePage pageContent={pageContent} preview={false} />);

      // test if card compoent is not rendered
      const page = document.querySelector('.page');
      expect(page).toBeNull();
    });
  });
});
