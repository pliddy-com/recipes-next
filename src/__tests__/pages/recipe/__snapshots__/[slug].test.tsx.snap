// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage in recipe/[slug].tsx when there is page content it renders the page 1`] = `
<div>
  <title>
    Site Title | Recipe Title
  </title>
  <meta
    content="Recipe abstract"
    name="description"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg page css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-a4d0r7-MuiTypography-root"
      >
        Recipe Title
      </h1>
      <div
        class="MuiBox-root css-0"
        data-testid="richtext"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
        >
          Description rich text
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-3ddw82"
    >
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="832px"
        imagesrcset="https://recipes.pliddy.com/url?w=832&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1664&fm=webp&q=75 2x"
        media="(min-width: 704px)"
        rel="preload"
      />
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="624px"
        imagesrcset="https://recipes.pliddy.com/url?w=624&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1248&fm=webp&q=75 2x"
        media="(min-width: 448px) and (max-width: 703px)"
        rel="preload"
      />
      <link
        as="image"
        href="https://recipes.pliddy.com/url"
        imagesizes="416px"
        imagesrcset="https://recipes.pliddy.com/url?w=416&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=832&fm=webp&q=75 2x"
        media="(max-width: 447px)"
        rel="preload"
      />
      <picture>
        <source
          media="(min-width: 704px)"
          sizes="832px"
          srcset="https://recipes.pliddy.com/url?w=832&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1664&fm=webp&q=75 2x"
        />
        <source
          media="(min-width: 448px) and (max-width: 703px)"
          sizes="624px"
          srcset="https://recipes.pliddy.com/url?w=624&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=1248&fm=webp&q=75 2x"
        />
        <source
          media="(max-width: 447px)"
          sizes="416px"
          srcset="https://recipes.pliddy.com/url?w=416&fm=webp&q=75 1x, https://recipes.pliddy.com/url?w=832&fm=webp&q=75 2x"
        />
        <img
          alt="Image description."
          height="300"
          loading="eager"
          src="https://recipes.pliddy.com/url"
          style="max-width: 100%; height: auto;"
          width="400"
        />
      </picture>
    </div>
    <div
      class="css-1lwbda4-MuiStack-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Ingredients
        </h2>
        <div
          class="css-nen11g-MuiStack-root"
        >
          <ul
            class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                  data-testid="RestaurantMenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  ingredient 1
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                  data-testid="RestaurantMenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  ingredient 2
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          EquipmentSection
        </h2>
        <ul
          class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="KitchenTwoToneIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5h2v3H8zm0 7h2v5H8zm-2 8h12v-9.02H6V20zm2-8h2v5H8v-5zM6 9h12V4H6v5zm2-4h2v3H8V5z"
                  opacity=".3"
                />
                <path
                  d="M18 2.01 6 2a2 2 0 0 0-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.11-.9-1.99-2-1.99zM18 20H6v-9.02h12V20zm0-11H6V4h12v5zM8 5h2v3H8zm0 7h2v5H8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                equipment 1
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="KitchenTwoToneIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5h2v3H8zm0 7h2v5H8zm-2 8h12v-9.02H6V20zm2-8h2v5H8v-5zM6 9h12V4H6v5zm2-4h2v3H8V5z"
                  opacity=".3"
                />
                <path
                  d="M18 2.01 6 2a2 2 0 0 0-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.11-.9-1.99-2-1.99zM18 20H6v-9.02h12V20zm0-11H6V4h12v5zM8 5h2v3H8zm0 7h2v5H8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                equipment 2
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="css-1okj3ks-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Instructions
        </h2>
        <div
          class="css-nen11g-MuiStack-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            Label
          </h3>
          <ul
            class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
            >
              <div
                class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
                  color="secondary"
                >
                  1
                </div>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Instructions item 1
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          Notes
        </h2>
        <ul>
          <li>
            note 1
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-0"
        data-testid="tags-section"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Tags
        </h2>
        <div
          class="css-180l46g-MuiStack-root"
          data-testid="tag-buttons"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-wmndcn-MuiButtonBase-root-MuiButton-root"
            href="/tag/tag-1"
            role="button"
            tabindex="0"
          >
            Tag 1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
